department-of-veterans-affairs/vets-website
# [CST] Incorrect PDF password error message not in viewport

## Original Issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/41989

## Expected behavior
The Claims Status Tool (CST) allows veterans to upload supporting documentation for their claim. One of the options that they have when uploading documentation is to use a password-protected PDF. When a veteran attempts to upload a password-protected PDF, a text input is provided for them to enter the password. If they enter the password incorrectly, when they click the 'Submit files for Review' button, an error alert is added to the screen informing them that they entered an incorrect password. This alert shows up quite a bit higher on the page than where the 'Submit files for Review' button is located, so it is expected that the browser window will scroll up the page so that the alert is visible.

## Current behavior
When a veteran attempts to upload a password-protected PDF and provides an incorrect password, the browser window scrolls up to the error alert, but only if the browser window is taller than a specific number of pixels (~1180px). If it is above this threshold, the browser window scrolls up the page to display the error alert. If the browser window's height is below that threshold, it does nothing.

## Your fix
I'm not 100% sure how the browser height affects the behavior described in the above sections, but this mainly seems to be a timing issue. When the veteran goes to upload a password-protected PDF file but has failed to provide the correct password, a `Notification` and an `Element` component is added to the DOM. The `Notification` is a component specific to the application, the `Element` component is provided by the `react-scroll` package. The `Element` component is right above the `Notification` component, and is what is actually being scrolled to when `scrollTo` is called. Because these components have to be added to the DOM, it seems that `scrollTo` is being called before the `Element` is in the DOM, therefore nothing happens. Wrapping the `scrollTo` call in a `setTimeout` seems to provide enough of a delay for things to work themselves out.
TLDR: This seems to be a timing issue with elements being added to the DOM, so wrapped the `scrollTo` in a `setTimeout`

## How has this been tested?
**NOTE:** You will need a password-protected PDF. [This](https://github.com/department-of-veterans-affairs/va.gov-team/files/8787544/TestPDF-password123.pdf) is one with password: `password123`
* Navigate to `/track-claims/your-claims` as a logged in user
* Select a claim from the claims list and click `View details`
* This takes you to the claims detail page, which should include a set of tabs towards the top
* Select the 'Files' tab
* Under 'Select files to upload', select 'Add Files'
* Select the password-protected PDF
* Under 'PDF password' enter an incorrect password
* Select any of the options under 'What type of document is this?'
* Check the box labeled 'The files I uploaded are supporting documents for this claim only.'
* Press the 'Submit files for Review' button
If your browser window is taller than (~1180px), the page should scroll up to an error alert when the button is pressed. If you button is not taller than the threshold, nothing should happen when pressing the button

## Screenshots
### Before
![before](https://user-images.githubusercontent.com/13838621/174162655-afceaceb-c258-455c-90ef-cee8baf1562d.gif)
### After
![after](https://user-images.githubusercontent.com/13838621/174162687-67d21fcb-2fe2-45e8-b9a3-461e10382f53.gif)

## Acceptance criteria
